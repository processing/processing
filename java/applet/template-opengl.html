<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
          "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
  <head>
    <!-- charset must remain utf-8 to be handled properly by Processing -->
    <meta http-equiv="content-type" content="text/html; charset=utf-8" />
    
    <title>@@sketch@@ : Built with Processing</title>
    
    <style type="text/css">
      /* <![CDATA[ */
	
		body {
  		  margin: 60px 0px 0px 55px;
		  font-family: verdana, geneva, arial, helvetica, sans-serif; 
		  font-size: 11px; 
		  background-color: #ddddcc; 
		  text-decoration: none; 
		  font-weight: normal; 
		  line-height: normal; 
		}
		 
		a          { color: #3399cc; }
		a:link     { color: #3399cc; text-decoration: underline; }
		a:visited  { color: #3399cc; text-decoration: underline; }
		a:active   { color: #3399cc; text-decoration: underline; }
		a:hover    { color: #3399cc; text-decoration: underline; }
	
		/* ]]> */
    </style>
    
  </head>
  <body>
    <div id="content">
      
      <div id="@@sketch@@_container">
	<applet code="org.jdesktop.applet.util.JNLPAppletLauncher"
		width="@@width@@"
		height="@@height@@"
		archive="@@archive@@,
			 http://jogamp.org/deployment/jogamp-current/jar/applet-launcher.jar,
			 http://jogamp.org/deployment/jogamp-current/jar/jogl.all.jar,
			 http://jogamp.org/deployment/jogamp-current/jar/gluegen-rt.jar">
	  <!--http://jogamp.org/deployment/webstart/jogl-demos/jogl-demos.jar-->
	  <param name="codebase_lookup" value="false" />
	  <param name="subapplet.classname" value="@@sketch@@" />
	  <!--<param name="subapplet.displayname" value="Pretty Name Here">-->
	  <param name="noddraw.check" value="true" />
	  <param name="progressbar" value="true" />
	  <param name="jnlpNumExtensions" value="1" />
	  <param name="jnlpExtension1"
		 value="http://jogamp.org/deployment/jogamp-current/jogl-all-awt.jnlp" />
	  <param name="java_arguments" value="-Dsun.java2d.noddraw=true" />
	  <!--<param name="jnlp_href" value="applet-gears.jnlp">-->
	  
	  <p>
	    <strong>
	      This browser does not have a Java Plug-in.
	      <br />
	      <a href="http://www.java.com/getjava" 
		 title="Download Java Plug-in">
		Get the latest Java Plug-in here.
	      </a>
	    </strong>
	  </p>

	</applet>
      </div>
   
      <p>
	@@description@@
      </p>
      
      <p>
	Source code: @@source@@
      </p>
      
      <p>
	Built with <a href="http://processing.org" title="Processing.org">Processing</a>
      </p>
    </div>
  </body>
</html>
