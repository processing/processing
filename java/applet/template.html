<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
  <head>
    <!-- charset must remain utf-8 to be handled properly by Processing -->
    <meta http-equiv="content-type" content="text/html; charset=utf-8" />

    <title>@@sketch@@ : Built with Processing</title>

    <style type="text/css">
      /* <![CDATA[ */
	
	body {
  	  margin: 60px 0px 0px 55px;
	  font-family: verdana, geneva, arial, helvetica, sans-serif; 
	  font-size: 11px; 
	  background-color: #ddddcc; 
	  text-decoration: none; 
	  font-weight: normal; 
	  line-height: normal; 
	}
		 
	a          { color: #3399cc; }
	a:link     { color: #3399cc; text-decoration: underline; }
	a:visited  { color: #3399cc; text-decoration: underline; }
	a:active   { color: #3399cc; text-decoration: underline; }
	a:hover    { color: #3399cc; text-decoration: underline; }

      /* ]]> */
    </style>    
  </head>

  <body>
    <div id="content">
      <div id="@@sketch@@_container">

	<!-- This version plays nicer with older browsers, 
	     but requires JavaScript to be enabled. 
	     http://java.sun.com/javase/6/docs/technotes/guides/jweb/deployment_advice.html
	     http://download.oracle.com/javase/tutorial/deployment/deploymentInDepth/ -->
	<script type="text/javascript"
		src="http://www.java.com/js/deployJava.js"></script>
	<script type="text/javascript">
	  /* <![CDATA[ */

	  var attributes = { 
            code: '@@sketch@@.class',
            archive: '@@archive@@',
            width: @@width@@, 
            height: @@height@@,
          };
          var parameters = { 
            image: 'loading.gif',
            centerimage: 'true',
          };
          var version = '1.5';
          deployJava.runApplet(attributes, parameters, version);

          /* ]]> */
        </script>
        
	<noscript> <div>
	  <!--[if !IE]> -->
	  <object classid="java:@@sketch@@.class" 
            	  type="application/x-java-applet"
            	  archive="@@archive@@"
            	  width="@@width@@" height="@@height@@"
            	  standby="Loading Processing software..." >
            
	    <param name="archive" value="@@archive@@" />
	    
	    <param name="mayscript" value="true" />
	    <param name="scriptable" value="true" />
	    
	    <param name="image" value="loading.gif" />
	    <param name="boxmessage" value="Loading Processing software..." />
	    <param name="boxbgcolor" value="#FFFFFF" />
	  <!--<![endif]-->

	    <!-- For more instructions on deployment, 
		 or to update the CAB file listed here, see:
		 http://java.sun.com/javase/6/webnotes/family-clsid.html
		 http://java.sun.com/javase/6/webnotes/install/jre/autodownload.html -->
	    <object classid="clsid:8AD9C840-044E-11D1-B3E9-00805F499D93"
		    codebase="http://java.sun.com/update/1.6.0/jinstall-6u20-windows-i586.cab"
		    width="@@width@@" height="@@height@@"
		    standby="Loading Processing software..."  >
	      
	      <param name="code" value="@@sketch@@" />
	      <param name="archive" value="@@archive@@" />
	      
	      <param name="mayscript" value="true" />
	      <param name="scriptable" value="true" />
	      
	      <param name="image" value="loading.gif" />
	      <param name="boxmessage" value="Loading Processing software..." />
	      <param name="boxbgcolor" value="#FFFFFF" />
	      
	      <p>
		<strong>
		  This browser does not have a Java Plug-in.
		  <br />
		  <a href="http://www.java.com/getjava" title="Download Java Plug-in">
		    Get the latest Java Plug-in here.
		  </a>
		</strong>
	      </p>
	      
	    </object>
	    
	  <!--[if !IE]> -->
	  </object>
	  <!--<![endif]-->

	</div> </noscript>

      </div>
      
      <p>
	@@description@@
      </p>
      
      <p>
	Source code: @@source@@
      </p>
      
      <p>
	Built with <a href="http://processing.org" title="Processing.org">Processing</a>
      </p>
    </div>
  </body>
</html>
