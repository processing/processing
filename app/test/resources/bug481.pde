import java.applet.Applet;
Class[] abc = new Class[]{Applet.class};