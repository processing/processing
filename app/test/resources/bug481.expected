import processing.core.*; 
import processing.data.*; 
import processing.event.*; 
import processing.opengl.*; 

import java.applet.Applet; 

import java.util.HashMap; 
import java.util.ArrayList; 
import java.io.File; 
import java.io.BufferedReader; 
import java.io.PrintWriter; 
import java.io.InputStream; 
import java.io.OutputStream; 
import java.io.IOException; 

public class bug481 extends PApplet {
    public void setup() {

Class[] abc = new Class[]{Applet.class};
        noLoop();
    }

    static public void main(String[] passedArgs) {
        String[] appletArgs = new String[] { "bug481" };
        if (passedArgs != null) {
          PApplet.main(concat(appletArgs, passedArgs));
        } else {
          PApplet.main(appletArgs);
        }
    }
}
